### ISSUE TITLE
Configuration manual missing non-default variants for `hover.memoryLayout.*` configs

### ISSUE THREAD

**Issue by chrisp60** [rust-lang/rust-analyzer#18251]
Hello, partial issue, partial question.

When configuring my LSP, I noticed that the below configurations show their **default** values, but do **not** show other valid values (variants).

I believe there are 3 configurations in the `memoryLayout` category that are guilty of this. 

The config assumes the end user is familiar with common representations of memory. I am sure most are, but maybe explicitness is preferred
 
# Config items without information on variants.
- [`memoryLayout.alignment`](https://rust-analyzer.github.io/manual.html#rust-analyzer.hover.memoryLayout.alignment) (Default "Hexadecimal")
- [`memoryLayout.size`](https://rust-analyzer.github.io/manual.html#rust-analyzer.hover.memoryLayout.size) (Default "Both")  
- [`memoryLayout.offset`](https://rust-analyzer.github.io/manual.html#rust-analyzer.hover.memoryLayout.offset) (Default "Hexadecimal")

These configs were deemed reasonable around [#14918](https://github.com/rust-lang/rust-analyzer/issues/14918).
[#14929](https://github.com/rust-lang/rust-analyzer/pull/14929) is when they were actually implemented. Interestingly, the latter of those PRs include a json file showing the variants for VSCode settings. Maybe it was forgotten for the reference on the website

If a PR is welcome, I can try to poke at it, I very unfamiliar with RA internals and standards, don't mind learning though

**Comment by darichey** [comment]
It looks like the manual doesn't document variants for _any_ config option, not just these. It doesn't look easily fixable, because the only place we have descriptions for the variants is in the `enumDescriptions` for the package.json: https://github.com/rust-lang/rust-analyzer/blob/0fb804acb375b02a3beeaceeb75b71969ef37b15/crates/rust-analyzer/src/config.rs#L3307-L3311

**Comment by chrisp60** [comment]
Hmm that's unfortunate. Maybe a decent half-measure exists by just stating where each variant can be found.

It's possible that this issue can just be ignored. I feel anyone that is tinkering with their lsp config outside of their editor's blanket is willing to go searching for the answers.

FWIW my editor does show the variants through a plugin (neovim with Mason). So maybe it is fine as it is.

A maintainer can weigh in / close this if they feel there are more pressing matters

**Comment by ChayimFriedman2** [comment]
Yes, the manual does not document options, and yes that is sad. It is not true though that package.json is the source of truth - it is itself generated from the config schema by the `generate_package_json_config()` test.
